metecho/api/model_mixins.py

```diff
@@ -162,3 +164,41 @@ def finalize_create_pr(self, *, error=None, originating_user_id):
             )
         else:
             self.notify_error(error=error, originating_user_id=originating_user_id)
+
+
+class SoftDeleteQuerySet(models.QuerySet):
+    def active(self):
+        return self.filter(deleted_at__isnull=True)
+
+    def delete(self):
+        soft_delete_child_class = getattr(self.model, "soft_delete_child_class", None)
+        if soft_delete_child_class:
+            parent = camel_to_snake(self.model.__name__)
+            soft_delete_child_class(None).objects.filter(
+                **{f"{parent}__in": self}
+            ).delete()
+        return self.active().update(deleted_at=timezone.now())
```

Comment:
@wlonk Note the change here, so that already-deleted items do not have `deleted_at` set again.